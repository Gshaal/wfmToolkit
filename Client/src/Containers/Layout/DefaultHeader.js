import React, { Component, Suspense} from "react";
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavbarText 
} from "reactstrap";
import HomeRoundedIcon from '@material-ui/icons/HomeRounded';
import EventNoteRoundedIcon from '@material-ui/icons/EventNoteRounded';
import BeachAccessRoundedIcon from '@material-ui/icons/BeachAccessRounded';
import SettingsRoundedIcon from '@material-ui/icons/SettingsRounded';
import PowerSettingsNewRoundedIcon from '@material-ui/icons/PowerSettingsNewRounded';
import Logo from '../../Assests/logo.gif';
import { NavLink } from 'react-router-dom';
import { connect } from "react-redux";
import * as actions from "../../Store/actions/index";


const Notification = React.lazy(() => import('./Notification'))

 class DefaultHeader extends Component {
  state = {
    isOpen: false,
    loading:true
  };

  toggle = () => this.setState({ isOpen: !this.state.isOpen });

  
  loading = () => (
    <div className="animated fadeIn pt-1 text-center">Loading...</div>
  );

  render() {
    return (
      <div>
      <Navbar  light expand="md">
        <NavbarBrand href="#">{"{WFM Toolkit}"}</NavbarBrand>
        <NavbarToggler onClick={()=>this.toggle()} />
        <Collapse isOpen={this.state.isOpen} navbar>
          <Nav className="mr-auto App-header" navbar>
            <NavItem >
              <NavLink to="/home"  exact={true} className="nav-link"  activeClassName="activeLink"> <HomeRoundedIcon style={{color:"#9A7AA0"}}  /> Home </NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/scheduler" exact={true} className="nav-link" activeClassName="activeLink">  <EventNoteRoundedIcon style={{color:"#9A7AA0"}}  /> Scheduler </NavLink>
            </NavItem>
            <NavItem className="navbar-logo">
            <img className="logo-img" alt="W" src={Logo} width="50" height="50" />
            </NavItem>
            <NavItem>
              <NavLink to="/holiday/me" exact={true} className="nav-link" activeClassName="activeLink"> <BeachAccessRoundedIcon style={{color:"#9A7AA0"}}  />Holiday</NavLink>
            </NavItem>
            <NavItem>
              <NavLink to="/setting" exact={true} className="nav-link" activeClassName="activeLink"> <SettingsRoundedIcon style={{color:"#9A7AA0"}}  />Settings</NavLink>
            </NavItem>
          </Nav>
          <Suspense fallback={this.loading()}>
          <Notification />
          </Suspense>
          <NavbarText onClick={(e) => this.props.onLogout()} style={{cursor:"pointer"}} title="logout" className="nav-link" ><PowerSettingsNewRoundedIcon style={{color:"#9A7AA0"}}  /></NavbarText>
        </Collapse>
      </Navbar>
    </div>
    );
  }
}


let mapDispatchToProps = (dispatch) => {
  return {
    onLogout: () => dispatch(actions.LogOut()),
  };
};

export default connect(null, mapDispatchToProps)(DefaultHeader);
