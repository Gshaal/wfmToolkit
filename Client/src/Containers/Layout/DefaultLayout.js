import React, { Component, Suspense } from "react";
import { Redirect, Route, Switch } from "react-router-dom";
import { Container } from "reactstrap";
import routes from "../../routes";

const DefaultHeader = React.lazy(()=>import('./DefaultHeader'))

export default class Layout extends Component {
  loading = () => (
    <div className="animated fadeIn pt-1 text-center">Loading...</div>
  );
  render() {
    return (
      <React.Fragment>
        <Suspense fallback={this.loading()}>
        <DefaultHeader />
        </Suspense>
        <main>
          <Container fluid>
            <Suspense fallback={this.loading()}>
              <Switch>
                {routes.map((route, idx) => {
                  return route.component ? (
                    <Route
                      key={idx}
                      path={route.path}
                      exact={route.exact}
                      name={route.name}
                      render={(props) => {
                        console.log("update route =>", this.props); //update the path
                        return <route.component {...props} />;
                      }}
                    />
                  ) : null;
                })}
                <Redirect
                    from="/login"
                    to={this.props.path ? this.props.path : "/home"}
                  />
                  <Redirect from="/" to="/home" />
              </Switch>
            </Suspense>
          </Container>
        </main>
      </React.Fragment>
    );
  }
}
