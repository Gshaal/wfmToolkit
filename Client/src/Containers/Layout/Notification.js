import React, { useState } from 'react';
import { Dropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
import NotificationsActiveRoundedIcon from '@material-ui/icons/NotificationsActiveRounded';

const Notification = (props) => {
  const [dropdownOpen, setDropdownOpen] = useState(false);

  return (
    <Dropdown direction="left" size="sm" className="notification-icon" isOpen={dropdownOpen} toggle={ () => setDropdownOpen(prevState => !prevState)} >
      <DropdownToggle tag="div"  >
        <NotificationsActiveRoundedIcon style={{color:"#9A7AA0"}} /> 
        <span className='badge badge-warning' id='lblCartCount'> 0 </span>
      </DropdownToggle>
      <DropdownMenu>
        <DropdownItem>
        <div className="text-center">Your have no new notifications</div>
        </DropdownItem>
      </DropdownMenu>
    </Dropdown>
  );
}

export default Notification;