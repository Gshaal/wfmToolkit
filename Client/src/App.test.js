
import React from 'react'
import Layout from './Containers/Layout/DefaultLayout'
import Enzyme,{shallow, configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({adapter:new Adapter()})

describe('<MainApp />', () => {
    it('renders the <layout /> components', () => {
      const wrapper = shallow(<Layout />);
      expect(wrapper).toMatchSnapshot();
    })
})