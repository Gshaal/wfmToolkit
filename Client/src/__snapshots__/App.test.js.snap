// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainApp /> renders the <layout /> components 1`] = `ShallowWrapper {}`;
